# 데이터 전처리

In [15]:
import pandas as pd 

In [16]:
df = pd.read_csv("C:/Users/whileduck/Desktop/train_activity.csv")

df.head()

EMAIL  activity_average_met  activity_cal_active  \
0  nia+279@rowan.kr               1.28125                  196   
1  nia+279@rowan.kr               1.25000                  145   
2  nia+279@rowan.kr               1.21875                  118   
3  nia+279@rowan.kr               1.28125                  180   
4  nia+279@rowan.kr               1.46875                  374   

   activity_cal_total activity_class_5min  activity_daily_movement  \
0                2251                 ...                     3353   
1                2159                 ...                     2516   
2                2140                 ...                     1716   
3                2240                 ...                     2791   
4                2559                 ...                     5393   

            activity_day_end         activity_day_start  activity_high  \
0  2020-10-20T03:59:59+09:00  2020-10-19T04:00:00+09:00              0   
1  2020-10-21T03:59:59+09:00  2020-10-20T04:00:00+09:00              0   
2  2020-10-22T03:59:59+09:00  2020-10-21T04:00:00+09:00              0   
3  2020-10-23T03:59:59+09:00  2020-10-22T04:00:00+09:00              0   
4  2020-10-24T03:59:59+09:00  2020-10-23T04:00:00+09:00              0   

   activity_inactive  ...  activity_score_move_every_hour  \
0                714  ...                             100   
1                719  ...                             100   
2                708  ...                              95   
3                705  ...                             100   
4                437  ...                             100   

   activity_score_recovery_time  activity_score_stay_active  \
0                           100                          54   
1                           100                          58   
2                           100                          55   
3                           100                          58   
4                           100                          89   

  activity_score_training_frequency  activity_score_training_volume  \
0                               100                              97   
1                                96                              96   
2                                71                              83   
3                                71                              59   
4                                40                              46   

   activity_steps  activity_total  \
0            4212             207   
1            3306             140   
2            2550             131   
3            4009             208   
4            8081             517   

             CONVERT(activity_class_5min USING utf8)  \
0  1/1/2/2/2/2/1/1/1/1/1/1/1/2/2/2/2/2/2/3/3/2/2/...   
1  1/1/1/2/2/2/2/2/2/2/2/2/1/1/1/1/1/2/3/3/3/3/2/...   
2  1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/1/2/3/3/3/...   
3  1/1/1/1/1/1/1/1/1/1/2/2/2/2/3/3/3/2/3/3/3/3/3/...   
4  1/1/1/2/2/2/2/2/2/1/1/1/3/2/2/1/2/2/2/3/3/2/2/...   

               CONVERT(activity_met_1min USING utf8)  DIAG_NM  
0  1.2/0.9/1/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/...      MCI  
1  0.9/0.9/1.3/1/0.9/0.9/1.1/0.9/1/0.9/1/0.9/1/0....      MCI  
2  0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0....      MCI  
3  0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0....      MCI  
4  0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0.9/0....      MCI  

[5 rows x 32 columns]

* EMAIL 은 인덱스 역할
* 각 사용자 별 약 2달간의 라이프로그 데이터가 존재함 

* Target data 는 DIAG_NM 컬럼으로 CN, MCI, DEM 존재 (무증상치매, 경도인지장애, 치매)

In [17]:
df['DIAG_NM'].value_counts()

CN     5781
MCI    3353
Dem     571
Name: DIAG_NM, dtype: int64

* Dementia 데이터는 571개로 약 14명의 사용자만 치매를 가지고 있음 

In [18]:
import numpy as np

email = np.array(df['EMAIL'])
target = np.array(df['DIAG_NM'])

target_dict = dict(np.c_[email,target]) # Grouping 이후 라벨링 하기 위한 딕셔너리 생성

In [19]:
drop_cols = df.describe(include = 'O').columns[1:] # 불필요한 행을 제거
                                                    # 해당 object 컬럼들은 모두 수치형 변수로 표현 되어 있음

print(drop_cols)

df = df.drop(drop_cols,axis = 1) 

Index(['activity_class_5min', 'activity_day_end', 'activity_day_start',
       'activity_met_1min', 'CONVERT(activity_class_5min USING utf8)',
       'CONVERT(activity_met_1min USING utf8)', 'DIAG_NM'],
      dtype='object')


In [20]:
df = df.groupby('EMAIL').mean().reset_index() # 그룹핑

In [22]:
df['target'] = df['EMAIL'].map(target_dict)

In [24]:
df = df.drop('EMAIL',axis = 1) # index 역할하던 email 제거 

In [25]:
df.head()

activity_average_met  activity_cal_active  activity_cal_total  \
0              1.262897           202.571429         2208.333333   
1              1.370299           217.477876         1773.504425   
2              1.390625           259.672414         2365.137931   
3              1.442555           401.382353         2503.882353   
4              1.388393           271.685714         2114.342857   

   activity_daily_movement  activity_high  activity_inactive  \
0              3659.730159       0.793651         615.142857   
1              5084.876106       0.256637         574.619469   
2              3877.741379       0.637931         632.482759   
3              7250.514706       0.382353         534.470588   
4              5053.200000       2.057143         484.085714   

   activity_inactivity_alerts  activity_low  activity_medium  \
0                    0.809524    146.428571        18.333333   
1                    0.539823    281.300885        15.398230   
2                    0.362069    328.827586         9.068966   
3                    0.176471    343.852941        30.970588   
4                    0.285714    326.485714        28.285714   

   activity_met_min_high  ...  activity_score  \
0               5.984127  ...       58.682540   
1               1.778761  ...       54.522124   
2               4.379310  ...       65.758621   
3               3.382353  ...       91.264706   
4               8.971429  ...       79.000000   

   activity_score_meet_daily_targets  activity_score_move_every_hour  \
0                          18.793651                       92.492063   
1                          11.601770                       95.566372   
2                          54.068966                       97.155172   
3                          92.147059                       99.117647   
4                          72.542857                       98.571429   

   activity_score_recovery_time  activity_score_stay_active  \
0                     99.698413                   66.269841   
1                     96.203540                   72.088496   
2                     99.862069                   64.965517   
3                     98.235294                   76.750000   
4                     99.885714                   83.142857   

   activity_score_training_frequency  activity_score_training_volume  \
0                          21.682540                       53.031746   
1                           6.371681                       46.530973   
2                          16.982759                       41.637931   
3                          83.750000                       93.058824   
4                          43.628571                       62.342857   

   activity_steps  activity_total  target  
0     4518.460317      165.555556     MCI  
1     6431.831858      296.955752      CN  
2     5868.758621      338.534483     MCI  
3     9444.338235      375.205882     MCI  
4     7211.600000      356.828571      CN  

[5 rows x 25 columns]

In [27]:
df.to_csv('C:/Users/whileduck/Desktop/Github/Dementia_Analysis/data/activity.csv',index = False)